pkg/autoops/domain/progressive_rollout.go

```diff
@@ -171,6 +171,62 @@ func (p *ProgressiveRollout) ExtractSchedules() ([]*autoopsproto.ProgressiveRoll
 	return nil, ErrProgressiveRolloutInvalidType
 }
 
+func (p *ProgressiveRollout) GetControlVariationID() (string, error) {
+	switch p.Type {
+	case autoopsproto.ProgressiveRollout_MANUAL_SCHEDULE:
+		c, err := unmarshalProgressiveRolloutManualClause(p.Clause)
+		if err != nil {
+			return "", err
+		}
+		// Try new field first, fallback to empty for backward compatibility
+		if c.ControlVariationId != "" {
+			return c.ControlVariationId, nil
+		}
+		return "", nil
+	case autoopsproto.ProgressiveRollout_TEMPLATE_SCHEDULE:
+		c, err := unmarshalProgressiveRolloutTemplateClause(p.Clause)
+		if err != nil {
+			return "", err
+		}
+		if c.ControlVariationId != "" {
+			return c.ControlVariationId, nil
+		}
+		return "", nil
```

Comment:
Returning the `c.ControlVariationId` directly is the same thing, whether it is empty or not.
```suggestion
		return c.ControlVariationId, nil
```